go-controller/pkg/ovn/gateway_init.go

```diff
@@ -495,11 +495,13 @@ func addPolicyBasedRoutes(nodeName, mgmtPortIP string, hostIfAddr *net.IPNet) er
 	}
 
 	if config.Gateway.Mode == config.GatewayModeLocal {
-		ipMask := hostIfAddr.IP.Mask(hostIfAddr.Mask)
-		hostNet := &net.IPNet{IP: ipMask, Mask: hostIfAddr.Mask}
+		var matchDst string
 		// Local gw mode needs to use DGP to do hostA -> service -> hostB
-		matchStr = fmt.Sprintf("%s.src == %s && %s.dst == %s /* inter-%s */",
-			l3Prefix, mgmtPortIP, l3Prefix, hostNet.String(), nodeName)
+		for _, clusterSubnet := range config.Default.ClusterSubnets {
+			matchDst += fmt.Sprintf(" && %s.dst != %s", l3Prefix, clusterSubnet.CIDR)
```

Comment:
I don't follow on why we add logical router policies with a destination IP being the cluster subnet? The cluster subnet is the subnet for regular pods, not nodes/`hostnetwork: true`-pods. Correct me if I am wrong here, but we need all node IPs in the destination filter?

Reply:
Which, by the way, if it is the case: becomes an issue on large clusters, as discussed here: https://ovn-org.slack.com/archives/C010SQ5FSNL/p1600188282180900

Reply:
Actually looping back on the last comment: we need a function to determine the set of node subnets spanning the entire cluster. Like this we could simplify all of these logical router policies that risk to become to big on large clusters. 

Ex: on a random OCP cluster I am on right now, we have:

```
 ip4.dst == 10.0.0.0/24 /* inter-zzhao-8481-kt89k-master-0 */
 ip4.dst == 10.0.1.0/24 /* inter-zzhao-8481-kt89k-worker-centralus-1 */
 ip4.dst == 10.0.0.0/24 /* inter-zzhao-8481-kt89k-master-1 */ 
 ip4.dst == 10.0.0.0/24 /* inter-zzhao-8481-kt89k-master-2 */
 ip4.dst == 10.0.1.0/24 /* inter-zzhao-8481-kt89k-worker-centralus-2 */
```

A destination filter such as `ip4.dst == 10.0.1.0/24 || ip4.dst == 10.0.0.0/24` is all we need to account for all possible destinations for this type of traffic. This solution should also be applied to the logical router policies stemming from the egress IP setup (as that is affected by the same issue)

/cc @girishmg

Reply:
The code is a != match, meaning dest traffic that does not match on cluster subnet traffic

Reply:
My bad, didn't catch that...